edu.wpi.first.wpilib.plugins.riolog/src/netconsole3/RioConsole.java

```diff
@@ -0,0 +1,325 @@
+      }
+      bytesRead += nRead;
+    }
+    //logger.log("finished reading segment");
+    return tag;
+  }
+
+  private Thread listener;
+  private Thread sender;
+
+  private final RioConnector rioConnector;
+  private final Lock lock = new ReentrantLock();
+  private final Condition wakeupListener = lock.newCondition();
+  private Socket socket = null;
+  private final BlockingQueue<Message> messageQueue = new LinkedBlockingQueue<>();
+  private boolean autoReconnect = true;
+  private final AtomicBoolean cleanup = new AtomicBoolean(false);
+  private final AtomicBoolean reconnect = new AtomicBoolean(false);
+  private final AtomicBoolean discard = new AtomicBoolean(false);
+  private final AtomicBoolean paused = new AtomicBoolean(false);
+  private final AtomicBoolean showWarning = new AtomicBoolean(true);
+  private final AtomicBoolean showPrint = new AtomicBoolean(true);
+  private Consumer<Boolean> connectedCallback = null;
+
+  @Override
+  protected void finalize() {
+    stop();
+  }
+
+  public void stop() {
+    cleanup.set(true);
+    closeSocket();
+    if (listener != null) {
+      listener.interrupt();
+    }
+  }
+
+  public void reconnect() {
+    reconnect.set(true);
+    closeSocket();
+  }
+
+  public boolean getAutoReconnect() {
+    return autoReconnect;
+  }
+
+  public void setAutoReconnect(boolean value) {
+    lock.lock();
+    try {
+      autoReconnect = value;
+      if (value) {
+        wakeupListener.signal();
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public boolean getPaused() {
+    return paused.get();
+  }
+
+  public boolean setPaused(boolean value) {
+    return paused.getAndSet(value);
+  }
+
+  public boolean getDiscard() {
+    return discard.get();
+  }
+
+  public boolean setDiscard(boolean value) {
+    return discard.getAndSet(value);
+  }
+
+  public boolean getShowWarning() {
+    return showWarning.get();
+  }
+
+  public boolean setShowWarning(boolean value) {
+    return showWarning.getAndSet(value);
+  }
+
+  public boolean getShowPrint() {
+    return showPrint.get();
+  }
+
+  public boolean setShowPrint(boolean value) {
+    return showPrint.getAndSet(value);
+  }
+
+  public void putDummyMessage() {
+    try {
+      messageQueue.put(new DummyMessage());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    }
+  }
+
+  private void setSocket(Socket socket) {
+    lock.lock();
+    this.socket = socket;
+    lock.unlock();
+  }
+
+  public void setConnectedCallback(Consumer<Boolean> callback) {
+    lock.lock();
+    this.connectedCallback = callback;
+    lock.unlock();
+  }
+
+  public BlockingQueue<Message> getMessageQueue() {
+    return messageQueue;
+  }
+
+  private static byte[] emptyFrame = new byte[] {0, 0};
+
+  private DataInputStream connect(int team) {
+    logger.log("connecting to team " + team);
+    Socket mySocket;
+    try {
+      mySocket = rioConnector.connect(team);
+    } catch (InterruptedException e) {
```

Comment:
Reset the thread's interrupted flag.
`Thread.currentThread().interrupt()`

Explanation:
http://www.yegor256.com/2015/10/20/interrupted-exception.html